backend/src/main/java/gov/cdc/usds/simplereport/service/model/UserInfo.java

```diff
@@ -27,16 +30,25 @@ public UserInfo(ApiUser user, Optional<OrganizationRoles> orgwrapper, boolean is
     this.roles =
         orgwrapper.map(OrganizationRoles::getGrantedRoles).orElse(Set.of()).stream()
             .collect(Collectors.toList());
-    Optional<OrganizationRole> effectiveRole =
-        orgwrapper.flatMap(OrganizationRoles::getEffectiveRole);
-    this.roleDescription = buildRoleDescription(effectiveRole, isAdmin);
-    effectiveRole.map(OrganizationRole::getGrantedPermissions).ifPresent(permissions::addAll);
+    Optional<Set<OrganizationRole>> effectiveRoles = orgwrapper.map(s -> s.getEffectiveRoles());
+    this.roleDescription = buildRoleDescription(effectiveRoles, isAdmin);
+    effectiveRoles
+        .map(s -> PermissionHolder.getPermissionsFromRoles(s))
+        .ifPresent(permissions::addAll);
+    this.facilities =
+        orgwrapper.map(OrganizationRoles::getFacilities).orElse(Set.of()).stream()
+            .collect(Collectors.toList());
     this.isAdmin = isAdmin;
   }
 
-  private String buildRoleDescription(Optional<OrganizationRole> role, boolean isAdmin) {
-    if (role.isPresent()) {
-      String desc = role.get().getDescription();
+  private String buildRoleDescription(Optional<Set<OrganizationRole>> roles, boolean isAdmin) {
+    if (roles.isPresent()) {
+      String desc =
```

Comment:
To this point on the frontend we have been using the `roleDescription` field as a ENUM to map to one of the following roles:

```
"Admin user" = Admin (full permissions)
"Standard user" = "Standard user (manage results and profiles)
"Test-entry user" = "Entry only (conduct tests)"
```

But with this PR is appears the `roleDescription` will now return a pipe-separated list in a string, so I'll get something like: `roleDescription: "All-facility-access user | Test-entry user"`

I can try to parse the user role from this string, but  want to make sure it's necessary first. It appears `ACCESS_ALL_FACILITIES ` will already be in the `permissions` list.

I'm also having trouble conceptualizing the difference between the `Member` and `User` role types. Can you help me understand the difference between the two?

Reply:
`Member` is a bare-bones role that offers no permissions to the user but allows us to mark them as being part of an Organization in Okta without automatically granting them permissions to a particular feature set. `User` is our traditional "I can edit patient data and view results in this org" role.

Reply:
Is there any way through our app to make someone just a "member"? Just trying to see if I need to accommodate this user type in the UI or if it's unlikely to exist.

Reply:
In summary:
addUser/addUserToCurrentOrg/updateUserRole: takes one role -- make it one of ADMIN, ENTRY_ONLY, USER (as we're already doing)
updateUserPrivileges: takes multiple roles -- include one of ADMIN, ENTRY_ONLY, USER and optionally also ALL_FACILITIES

Reply:
this is admittedly a downside of squeezing ALL_FACILITIES into the same concept as ENTRY_ONLY, USER, and ADMIN -- once roles become a "choose several but please include some of these" as opposed to a "choose one and only one" model, other things become a bit more brittle. cc'ing @jeskew-gov and @benwarfield-usds who might have thoughts.